fix: Update French translations.

arkhi opened the pull request:
- Replace `...` with `…` for all languages.
- Have consistent ` "LANG" : "…",` pattern strings.
- Fix some typos in English (dowload, reasign).

In French:

- Replace `'` (quote) with `’` (apostrophe).
- Replace `“` and `”` with `«` and `»`.
- Replace `-` (hyphens) with `‑` ([non‑breaking hyphens](https://unicode-table.com/en/2011/)).
- Fix some grammar issues.
- Replace `Editer` with `Modifier`.
- Replace `Espacement des lettres` with [`Crénage`](https://fr.wikipedia.org/wiki/Cr%C3%A9nage); shorter term for “kerning”.
- `!`, `?`, `;`, `:`, `«` and `»` take a [non‑breaking space](https://unicode-table.com/en/202F/) before or after them in French.
- _Accord de proximité_ on one occasion. (masculine + feminine + adjective = feminine adjective).
- “Soulignage” and “Barré” (I looked at LibreOffice to see how they were doing it).
- Consistent use of “Êtes‑vous sûr de vouloir ”.
- bibliothèque partagée: Bibliothèque Partagée.
- « Mise à jour » to use a noun that is not gender ambiguous.
- “Disposition” changed to “Mise en page” (could be “Composition”, although more ambiguous with other terms).
- Hauteur de ligne: Interlignage.
- Crénage: [Interlettrage](https://fr.wikipedia.org/wiki/Interlettre) which is more what a typographer would do based on the existing kerning of the font.
- Première lettre en majuscule: Premières Lettres en [Capitales](https://fr.wikipedia.org/wiki/Capitale_et_majuscule) (to illustrate the result).
- Quitter: Se déconnecter (clearer about the outcome of the action).
- Use of “a” for the title and “the” for the confirmation.
- Couche: Calque.

arkhi committed:
Add Narrow No-Break Space in French.

See https://unicode-table.com/en/202F/ for the character to add before `!`, `?`, `;`, `:` and `»`, and after `«`.

madmath03 commented on frontend/resources/locales.json:
```diff
@@ -21,7 +21,7 @@
     "translations" : {
       "en" : "Create demo account",
-      "fr" : "Vous voulez juste essayer?",
+      "fr" : "Vous voulez juste essayer ?",
```
I made a mistake when I first translated this (wrong phrase). Should be:
```suggestion
      "fr" : "Créer un compte de démonstration",
```

arkhi replied:
This is addressed in https://github.com/penpot/penpot/pull/591/commits/341d6a1b008a1e813e89ba128e3fb5210b76fde2.

madmath03 reviewed (changes requested):
just one small fix of a mistake I originnaly made, but seems good overall.

Thank you very much @arkhi

arkhi committed:
fix: Typography

Replace `...` with `…` for all languages.

In French:

- Replace `'` (quote) with `’` (apostrophe).
- Replace `“` and ” with `«` and `»`.
- Replace `-` (hyphens) with `‑` (non‑breaking hyphens).
- Fix a few grammar issues.
- Replace `Editer` with `Modifier`.
- Replace `Espacement des lettres` with `Crénage`; shorter term for “kerning”.
- Add accents on uppercase letters.

arkhi committed:
Fix a string in French.

arkhi committed:
Missed two replacements in French.

madmath03 reviewed (changes requested):
If we go with changing `“` by `«` and `»`, then there are 2 others places to modify: `"Aucune correspondance pour « %s »"`

```json
  "dashboard.no-matches-for" : {
    "used-in" : [ "src/app/main/ui/dashboard/search.cljs:54" ],
    "translations" : {
      "en" : "No matches found for “%s“",
      "fr" : "Aucune correspondance pour “%s“",
      "ru" : "Совпадений для “%s“ не найдено",
      "es" : "No se encuentra “%s“"
    }
  },
```

```json
  "workspace.libraries.no-matches-for" : {
    "used-in" : [ "src/app/main/ui/workspace/libraries.cljs:122" ],
    "translations" : {
      "en" : "No matches found for “%s“",
      "fr" : "Aucune correspondance pour “%s“",
      "ru" : "Совпадений для “%s“ не найдено",
      "es" : "No se encuentra “%s“"
    }
  },
```

There are also 2 other occurences of `Espacement des lettres` that need to be changed to `Crénage`:
```json
  "handoff.attributes.typography.letter-spacing" : {
    "used-in" : [ "src/app/main/ui/handoff/attributes/text.cljs:144" ],
    "translations" : {
      "en" : "Letter Spacing",
      "fr" : "Espacement des lettres",
      "es" : "Espaciado de letras"
    }
  },
```
  
```json
  "workspace.options.text-options.letter-spacing" : {
    "used-in" : [ "src/app/main/ui/workspace/sidebar/options/typography.cljs:154" ],
    "translations" : {
      "en" : "Letter Spacing",
      "fr" : "Espacement des lettres",
      "ru" : "Межсимвольный интервал",
      "es" : "Espaciado entre letras"
    }
  },
```

arkhi committed:
Add missing changes:

- French typographic quotes.
- Crénage

Addresses https://github.com/penpot/penpot/pull/591#pullrequestreview-585038080

madmath03 reviewed (changes requested):
Unless I made a mistake, `"Aucune correspondance pour “%s“"` is still there and needs to be udpated.

arkhi committed:
Update locales:

- Fix some typos in English (dowload, reasign).

- Fix some grammar.
- _Accord de proximité_ on one occasion. (masculine + feminine + adjective = feminine adjective).
- “Soulignage” and “Barré” (I looked at LibreOffice to see how they were doing it).
- Consistent use of “Êtes‑vous sûr de vouloir ”.
- bibliothèque partagée: Bibliothèque Partagée.
- « Mise à jour » to use a noun that is not gender ambiguous.
- “Disposition” changed to “Mise en page” (could be “Composition”, although more ambiguous with other terms).
- Hauteur de ligne: Interlignage.
- Crénage: [Interlettrage](https://fr.wikipedia.org/wiki/Interlettre) which is more what a typographer would do based on the existing kerning of the font.
- Première lettre en majuscule: Premières Lettres en [Capitales](https://fr.wikipedia.org/wiki/Capitale_et_majuscule) (to illustrate the result).
- Quitter: Se déconnecter (clearer about the outcome of the action).
- Use of “a” for the title and “the” for the confirmation.
- Couche: Calque.

arkhi committed:
Update a missed string for consistency.

madmath03 reviewed (commented):
@arkhi Thank you very much on all this work ❤️ 
Regarding your potential changes you mentioned:

> adresse e‑mail: e‑mail?

I would prefer to keep "_adresse e‑mail_" to not confuse it with an e-mail sent.

> Information de l’équipe: À propos de l’équipe?

"à propos" is mostly used when you want to describe something. Here, this key is just for the name and logo of the team.
I think we can keep "_Information_"

> L’ancien mot de passe est incorrect: Le mot de passe existant est incorrect

Hm... Given the context (Settings > Password > Old password), I think "_l'ancien_" fits better, or maybe "_Le mot de passe actuel est incorrect_" (the current password is incorrect) ?

> Trait: Contour?

Depends on the context here. If we are on a group or other object, sure, but if the selection is a Path or Curve, it is actually the "_Trait_" properties.

> Tiret: Traitillé? (Swiss neologism… but that could work ; “Tirets” otherwise?)

I would vote for "_Tirets_"

> Largeur: Épaisseur? (for stroke)

For stroke, you're definitely right :+1:.

> Ombres de la sélection: Ombres sur sélection?

I would either keep it as is or also change `workspace.options.text-options.title-group` and `workspace.options.text-options.title-selection` to have the same format.

> Seems like workspace.options.grid.column does not have the right plural/singular logic but I can’t tell if the key or the translations are wrong.

Would make more sense to rename the key to `workspace.options.grid.columns` (Translation seems good).

> Not sure about the gender for workspace.options.none.

This is used to specify the destination for prototyping, which points to a "Plan de travail" (Artboard). I think the gender is good.

@niwinz I guess we can wait for @arkhi answers on his suggestions, but after it should be okay for review.

arkhi committed:
[L10N] Update some French terms.
